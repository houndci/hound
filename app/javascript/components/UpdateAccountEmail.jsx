import React from 'react';

import App from './UpdateAccountEmail/components/App';

export default class UpdateAccountEmail extends React.Component {
  render() {
    return(
      <App {...this.props} />
    );
  }
}

