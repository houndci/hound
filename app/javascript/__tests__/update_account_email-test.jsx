import UpdateAccountEmail from '../components/UpdateAccountEmail/components/App';

it('renders appropriately', () => {
  const wrapper = shallow(
    <UpdateAccountEmail
      addressChanged={false}
    />
  );
  expect(wrapper).toMatchSnapshot();
});

it('renders appropriately on address change', () => {
  const wrapper = shallow(
    <UpdateAccountEmail
      addressChanged={true}
    />
  );
  expect(wrapper).toMatchSnapshot();
});

it('sets state appropriately on email input', () => {
  const wrapper = mount(
    <UpdateAccountEmail
      addressChanged={false}
    />
  );

  const emailInput = wrapper.find('#email_address');
  emailInput.simulate('change', {target: {value: "new text"}});

  expect(wrapper.state('emailInput')).toBe('new text');
  expect(wrapper.state('addressChanged')).toBe(null);
});
