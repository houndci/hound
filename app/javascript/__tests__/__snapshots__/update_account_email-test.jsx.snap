// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders appropriately 1`] = `
<article
  className="account-details"
>
  <h3>
    Update account settings
  </h3>
  <form>
    <div
      className="form-group"
    >
      <label>
        Email address for receipts
      </label>
      <input
        id="email_address"
        onChange={[Function]}
        type="email"
      />
      <UpdateAccountEmailMessage
        addressChanged={null}
      />
    </div>
    <div
      className="form-actions"
    >
      <button
        className="repo-toggle"
        onClick={[Function]}
      >
        Update Email
      </button>
    </div>
  </form>
</article>
`;

exports[`renders appropriately on address change 1`] = `
<article
  className="account-details"
>
  <h3>
    Update account settings
  </h3>
  <form>
    <div
      className="form-group"
    >
      <label>
        Email address for receipts
      </label>
      <input
        id="email_address"
        onChange={[Function]}
        type="email"
      />
      <UpdateAccountEmailMessage
        addressChanged={null}
      />
    </div>
    <div
      className="form-actions"
    >
      <button
        className="repo-toggle"
        onClick={[Function]}
      >
        Update Email
      </button>
    </div>
  </form>
</article>
`;
